src/towncrier/_git.py

```diff
@@ -5,12 +5,26 @@
 
 import os
 
-from subprocess import STDOUT, call, check_output
+from subprocess import STDOUT, CalledProcessError, call, check_output
 
 
 def remove_files(fragment_filenames: list[str]) -> None:
-    if fragment_filenames:
-        call(["git", "rm", "--quiet"] + fragment_filenames)
+    if not fragment_filenames:
+        return
+
+    # Filter out files that are unknown to git
+    try:
+        git_fragments = check_output(
+            ["git", "ls-files"] + fragment_filenames, encoding="utf-8"
+        ).split("\n")
+        git_fragments = [os.path.abspath(f) for f in git_fragments if os.path.isfile(f)]
```

Comment:
Minor comment

can we have this outside of try/except ?

Are we expecting this line to raise `CallerProcessError` ?